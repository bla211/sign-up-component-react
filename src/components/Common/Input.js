import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Input extends Component {
    render(){
        const { placeholder, className, val, handleChange, type, autofocus, onclick, isError, data, onError} = this.props;
        
        let classNames;
        if(className !== undefined){
            classNames = className.join(' ');
        }

        let handleKeyPress = (event) => {
            if(event.key == 'Enter'){
                if(!isError){
                    onclick(data);
                }
                else{
                    onError(true);
                }
            }
        }

        return(
            <input type={ type } placeholder={ placeholder } 
                className={ classNames } value={ val } onChange={ (event) => handleChange(event.target.value) }
                onKeyPress={ (event) => handleKeyPress(event) } autoFocus={ autofocus }
            />
        )
    }
}

Input.propTypes = {
    placeholder: PropTypes.string,
    className: PropTypes.array,
    val: PropTypes.string,
    handleChange: PropTypes.func,
    type: PropTypes.string,
    autofocus: PropTypes.bool,
    
    onclick: PropTypes.func,
    data: PropTypes.string,
    isError: PropTypes.bool,
    onError: PropTypes.func
  };

export default Input;