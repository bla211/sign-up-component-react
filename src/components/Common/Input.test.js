import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Input from './Input';

configure({ adapter: new Adapter() });

describe('Test Input component', () => {
    test('Input renders HTML when passed props', () => {
        const placeholder = 'placeholder text';
        const className = ['className'];
        const val = 'value';
        const type = 'text';
        const autofocus = false;
        const wrapper = shallow(<Input placeholder={placeholder} className={className} val={val} type={type} autofocus={autofocus}/>);
        expect(wrapper).toMatchSnapshot()
    });

    test('Input passes input value on input change', () => {
        const className = ['className'];
        const handleChange = jest.fn();
        const val = ''; 
        const wrapper = shallow(<Input val={val} handleChange={handleChange} className={className}/>);
        wrapper.find('input').simulate('change', { target: { value: 'value' } })
        expect(handleChange).toHaveBeenCalledWith('value');
    });

    test('Input fires onError on enter press if "isError === true"', () => {
        const className = ['className'];
        const onError = jest.fn(); 
        const isError = true;
        const wrapper = shallow(<Input onError={onError} isError={isError} className={className}/>);
        wrapper.find('input').simulate('keypress', { key: 'Enter', keyCode: 13, which: 13 });
        expect(onError).toHaveBeenCalled();
    });
});