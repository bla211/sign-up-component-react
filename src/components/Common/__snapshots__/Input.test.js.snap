// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input component Input renders properly when passed props 1`] = `
<input
  autoFocus={false}
  className="className"
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="placeholder text"
  type="text"
  value="value"
/>
`;
